test/torch_np/test_random.py

```diff
@@ -2,54 +2,127 @@
 
 """Light smoke test switching between numpy to pytorch random streams.
 """
+from contextlib import contextmanager
+
+import numpy as _np
 import pytest
 
 import torch._numpy as tnp
 from torch._numpy.testing import assert_equal
 
 
-def test_uniform():
-    r = tnp.random.uniform(0, 1, size=10)
-
-
-def test_shuffle():
-    x = tnp.arange(10)
-    tnp.random.shuffle(x)
-
-
-def test_numpy_global():
-    tnp.random.USE_NUMPY_RANDOM = True
-    tnp.random.seed(12345)
-    x = tnp.random.uniform(0, 1, size=11)
-
-    # check that the stream is identical to numpy's
-    import numpy as _np
-
-    _np.random.seed(12345)
-    x_np = _np.random.uniform(0, 1, size=11)
-
-    assert_equal(x, tnp.asarray(x_np))
-
-    # switch to the pytorch stream, variates differ
-    tnp.random.USE_NUMPY_RANDOM = False
-    tnp.random.seed(12345)
-
-    x_1 = tnp.random.uniform(0, 1, size=11)
-    assert not (x_1 == x).all()
-
-
-def test_wrong_global():
+@contextmanager
+def control_stream(use_numpy=False):
+    oldstate = tnp.random.USE_NUMPY_RANDOM
+    tnp.random.USE_NUMPY_RANDOM = use_numpy
     try:
-        oldstate = tnp.random.USE_NUMPY_RANDOM
-
-        tnp.random.USE_NUMPY_RANDOM = "oops"
-        with pytest.raises(ValueError):
-            tnp.random.rand()
-
+        yield
     finally:
         tnp.random.USE_NUMPY_RANDOM = oldstate
 
 
+@pytest.mark.parametrize("use_numpy", [True, False])
+@pytest.mark.parametrize(
+    "name, arg",
+    [
+        ("normal", ()),
+        ("rand", ()),
+        ("randint", (0, 5)),
+        ("randn", ()),
+        ("random", ()),
+        ("random_sample", ()),
+        ("sample", ()),
+        ("uniform", ()),
+    ],
+)
```

Comment:
You can simply put the `tnp.random.*` functions here. Then you can bind `partial(tnp.random.randint, 0, 5)` and remove the `arg` parameter. If you do this, the `test_array` can be written much clearer as
```python
def test_array(self, fn, use_numpy):
    with control_stream(use_numpy):
        if name in ("rand", "randn"):
            r = fn(10)
        else:
            r = fn(size=10)
    assert isinstance(r, tnp.ndarray)
```
and in the scalar one you can simply call `fn()`.